src/sentry/sentry_metrics/kafka_metrics_interface.py

```diff
@@ -0,0 +1,138 @@
+from concurrent.futures import Future
+from datetime import datetime
+from typing import Any, Mapping, Optional, Sequence, Union
+
+from arroyo import Topic
+from arroyo.backends.kafka import KafkaPayload, KafkaProducer, build_kafka_configuration
+
+# from arroyo.types import BrokerValue
+from django.conf import settings
+
+from sentry.sentry_metrics.metrics_interface import GenericMetricsBackend
+from sentry.sentry_metrics.use_case_id_registry import UseCaseID
+from sentry.utils import json
+from sentry.utils.kafka_config import get_kafka_producer_cluster_options
+
+
+def build_mri(metric_name: str, type: str, use_case_id: UseCaseID, unit: Optional[str]) -> str:
+    mri_unit = "none" if unit is None else unit
+    return f"{type}:{use_case_id.value}/{metric_name}@{mri_unit}"
+
+
+# def set_future(f: Future[BrokerValue[KafkaPayload]]) -> Union[Future[None], Future[Any]]:
+#     new_future: Union[Future[None], Future[Any]] = Future()
+
+#     if f.exception() is not None:
+#         new_future.set_exception(f.exception())
+#     else:
+#         new_future.set_result(f.result)
+
+#     return new_future
+
+
+class KafkaMetricsBackend(GenericMetricsBackend):
+    def __init__(self) -> None:
+
+        self.kafka_topic = settings.KAFKA_INGEST_PERFORMANCE_METRICS
+
+        cluster_name = settings.KAFKA_TOPICS[self.kafka_topic]["cluster"]
+        producer_config = get_kafka_producer_cluster_options(cluster_name)
+        self.producer = KafkaProducer(build_kafka_configuration(default_config=producer_config))
+
+    def counter(
+        self,
+        use_case_id: UseCaseID,
+        org_id: int,
+        project_id: int,
+        metric_name: str,
+        value: Union[int, float],
+        tags: Mapping[str, str],
+        unit: Optional[str] = None,
+        retention_days: Optional[int] = 90,
+    ) -> Optional[Future[Any]]:
```

Comment:
I don't think `Any` is sufficient here. This is an API we expect multiple products to use. We should be clearer about what type we are returning to users.

Reply:
Sure, I made the type for the Kafka implementation more specific.

Reply:
We shouldn't return a KafkaPayload though, the signature should be the same regardless of the backend picked. That is, we shouldn't return Kafka specific stuff to the user.

Reply:
Ah, so are you saying that all of the backend implementations should have the same return type? I've kept the abstract class's return type to be Any and thought that maybe each implementation could return something more specific. Do you have something else in mind? One other option would be to make all implementations + abstract class return `Future[None]`, but at that point what is the value of a `Future` to a user?